hopsworks-common/src/main/java/io/hops/hopsworks/common/jupyter/JupyterController.java

```diff
@@ -92,25 +94,31 @@
   private HdfsUsersFacade hdfsUsersFacade;
 
   @TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
-  public void convertIPythonNotebook(String hdfsUsername, String path, Project project, String outPath)
+  public void convertIPythonNotebook(String hdfsUsername, String notebookPath, Project project, String pyPath)
       throws ServiceException {
 
+    String conversionDir = DigestUtils.sha256Hex(Integer.toString(ThreadLocalRandom.current().nextInt()));
+    notebookPath = notebookPath.replace(" ", "\\ ");
+    pyPath = pyPath.replace(" " , "\\ ");
+
     String prog = settings.getHopsworksDomainDir() + "/bin/convert-ipython-notebook.sh";
     ProcessDescriptor processDescriptor = new ProcessDescriptor.Builder()
         .addCommand(prog)
-        .addCommand(path)
+        .addCommand(notebookPath)
         .addCommand(hdfsUsername)
         .addCommand(settings.getAnacondaProjectDir(project))
-        .addCommand(outPath)
-        .ignoreOutErrStreams(true)
+        .addCommand(pyPath)
+        .addCommand(conversionDir)
         .setWaitTimeout(60l, TimeUnit.SECONDS) //on a TLS VM the timeout needs to be greater than 20s
         .build();
 
     LOGGER.log(Level.FINE, processDescriptor.toString());
     try {
       ProcessResult processResult = osProcessExecutor.execute(processDescriptor);
-
       if (!processResult.processExited() || processResult.getExitCode() != 0) {
+        LOGGER.log(Level.SEVERE, "Failed to convert " + notebookPath
```

Comment:
@robzor92 this log message can be moved to the ServiceException message line 123

Reply:
Fixed